src/js/App/GlobalFilter/GlobalFilterMenu.js

```diff
@@ -0,0 +1,169 @@
+      .filter(
+        (item) =>
+          group.noFilter ||
+          (value && filterValueRegex.test(value)) ||
+          (label && filterValueRegex.test(label)) ||
+          (item.value && filterValueRegex.test(item.value)) ||
+          (item.label && filterValueRegex.test(item.label))
+      )
+      .map((item, index) => ({
+        ...item,
+        key: item.id || item.value || index,
+        value: String(item.value || item.id || index),
+        onClick: (event) => {
+          onChange(
+            event,
+            calculateSelected(type, value, item.value),
+            {
+              value,
+              label,
+              id,
+              type,
+              items,
+              ...group,
+            },
+            item,
+            value,
+            item.value
+          );
+        },
+      })),
+  }));
+  return result.filter(({ noFilter, items = [] }) => noFilter || items.length > 0);
+};
+
+const GlobalFilterMenu = (props) => {
+  const { filterBy, onFilter, groups = [], onChange, selectedTags } = props;
+  const [isOpen, setIsOpen] = useState(false);
+
+  const calculateSelected = (type, groupKey, itemKey) => {
+    const activeGroup = selectedTags[groupKey];
+    if (activeGroup) {
+      if (type !== groupType.radio && (activeGroup[itemKey] instanceof Object ? activeGroup[itemKey].isSelected : Boolean(activeGroup[itemKey]))) {
+        return {
+          ...selectedTags,
+          [groupKey]: {
+            ...(activeGroup || {}),
+            [itemKey]: false,
+          },
+        };
+      }
+
+      return {
+        ...selectedTags,
+        [groupKey]: {
+          ...(type !== groupType.radio ? activeGroup || {} : {}),
+          [itemKey]: true,
+        },
+      };
+    }
+
+    return {
+      ...selectedTags,
+      [groupKey]: {
+        [itemKey]: true,
+      },
+    };
+  };
+  const sanitizedFilterRegex = useMemo(() => {
+    try {
+      return new RegExp(filterBy, 'i');
+    } catch (err) {
+      return new RegExp(filterBy.replace(/[.*+?^${}()|[\]\\]/g, '\\$&'), 'i');
+    }
+  }, [filterBy]);
+
+  const menuItems = getMenuItems(groups, sanitizedFilterRegex, onChange, calculateSelected);
+  const menu = [
+    <div key="x" className="pf-c-menu ins-c-global-filter__menu" style={{ boxShadow: 'none' }}>
+      {menuItems.map(({ value, label, items }) => (
+        <MenuGroup key={value} label={label} value={value}>
+          <MenuList>
+            {items.map(({ value, label, onClick, ...props }) => (
+              <MenuItem key={value} onClick={onClick}>
+                <Checkbox
+                  className="ins-c-global-filter__checkbox"
+                  // eslint-disable-next-line react/prop-types
+                  id={props.id}
+                  // eslint-disable-next-line react/prop-types
+                  isChecked={!!Object.values(selectedTags).find((group = {}) => group[props.tagKey]?.isSelected)}
+                  label={label}
+                />
```

Comment:
This should change based on the `type`, however for the chrome's implementation it doesn't have to be here.

Reply:
Yes, I am aware. I think we can leave the different types for the FEC version.